system/jlib/jmisc.cpp

```diff
@@ -1082,3 +1082,33 @@ std::pair<std::string, unsigned> getDafileServiceFromConfig(const char *applicat
         throw makeStringExceptionV(-1, "dafilesrv '%s': external service port not defined", dafilesrvName.str());
     return externalService;
 }
+
+#if defined(__linux__)
+static unsigned envSetterHookId = 0;
+MODULE_INIT(INIT_PRIORITY_STANDARD)
+{
+    // initialize built-in HPCC environment variables from settings,
+    // such that they can be access from ECL from getEnv calls.
+    auto envSetterFunc = [](const IPropertyTree *oldComponentConfiguration, const IPropertyTree *oldGlobalConfiguration)
+    {
+        StringBuffer deploymentName;
+        if (isContainerized())
+            getGlobalConfigSP()->getProp("@deploymentName", deploymentName);
+        else
+            queryEnvironmentConf().getProp("deploymentName", deploymentName);
+        if (0 == deploymentName.length())
+            deploymentName.set("UNKNOWN");
+        setenv("HPCC_DEPLOYMENT", deploymentName, 1);
```

Comment:
ok, the original idea was to hook getEnv, so that it special cased based on some convention, and got the value from our own internal env. section.
Pity, but I'll change to follow that route vs using standard process env. variables.

Reply:
@ghalliday - please see revised scheme.